carma_wm/include/carma_wm/WMTestLibForGuidance.h

```diff
@@ -0,0 +1,458 @@
+#pragma once
+/*
+ * Copyright (C) 2020 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <gmock/gmock.h>
+#include <iostream>
+#include <carma_wm/CARMAWorldModel.h>
+#include <lanelet2_core/geometry/LineString.h>
+#include <lanelet2_traffic_rules/TrafficRulesFactory.h>
+#include <lanelet2_core/Attribute.h>
+#include <cav_msgs/RoadwayObstacle.h>
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/polygon.hpp>
+#include <tf2/LinearMath/Transform.h>
+#include <lanelet2_core/geometry/Polygon.h>
+#include <lanelet2_extension/regulatory_elements/DigitalSpeedLimit.h>
+#include <carma_wm/Geometry.h>
+/**
+ * This is a test library made for guidance unit tests. It includes following in general:
+ * - Helper functions to create the world from scratch or extend the world in getGuidanceTestMap()
+ * - addObstacle at a specified Cartesian or Trackpos point relative to specified lanelet Id
+ * - set route by giving series of lanelet Id in the map (setRouteById)
+ * - set speed of entire road (setSpeedLimit)
+ * - getGuidanceTestMap gives a simple one way, 3 lane map (25mph speed limit) with one static prebaked obstacle and 
+ *      4 lanelets in a lane (if 2 stripes make up one lanelet):
+ *
+ *        |1203|1213|1223|
+ *        | _  _  _  _  _|
+ *        |1202| Ob |1222|
+ *        | _  _  _  _  _|
+ *        |1201|1211|1221|    num   = lanelet id hardcoded for easier testing
+ *        | _  _  _  _  _|    |     = lane lines
+ *        |1200|1210|1220|    - - - = Lanelet boundary
+ *        |              |    O     = Default Obstacle
+ *        ****************
+ *           START_LINE
+ * - NOTE: please look at unit test on example how to use them
```

Comment:
It would be better if you could add some examples to the README similar to how is done for the basic carma_wm usage